src/components/loader-spinner/loader-spinner.spec.tsx

```diff
@@ -0,0 +1,244 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { LoaderSpinner, LoaderSpinnerSizes } from ".";
+import useMediaQuery from "../../hooks/useMediaQuery";
+import { LOADER_SPINNER_SIZE_PARAMS } from "./loader-spinner.config";
+
+jest.mock("../../hooks/useMediaQuery", () => {
+  return {
+    __esModule: true,
+    default: jest.fn().mockReturnValue(true),
+  };
+});
+
+describe("LoaderSpinner", () => {
+  describe("when the component is rendered", () => {
+    it("the component wrapper renders with a role of status", () => {
+      const { getByRole } = render(<LoaderSpinner />);
+      const wrapperElement = getByRole("status");
+
+      expect(wrapperElement).toBeInTheDocument();
+    });
+
+    it("the svg spinner circle renders", () => {
+      const { container } = render(<LoaderSpinner />);
+      const svgCircleElement = container.querySelector("circle")!;
+
+      expect(svgCircleElement).toBeInTheDocument();
+    });
+
+    it("the spinner label renders", () => {
+      const { container } = render(<LoaderSpinner />);
+      expect(
+        container.querySelector('[data-element="spinner-label"]')
+      ).toHaveTextContent("Loading...");
+    });
+  });
+
+  describe("when custom props are passed", () => {
+    it.each(["XS", "S", "M", "L", "XL"] as LoaderSpinnerSizes[])(
```

Comment:
suggestion (non-blocking): you could store this in a typed array and then re-use it rather than casting to the type etc